server/src/main/java/org/opensearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -194,6 +227,40 @@ protected TransportReplicationAction(
         String executor,
         boolean syncGlobalCheckpointAfterOperation,
         boolean forceExecutionOnPrimary
+    ) {
+        this(
+            settings,
+            actionName,
+            transportService,
+            clusterService,
+            indicesService,
+            threadPool,
+            shardStateAction,
+            actionFilters,
+            requestReader,
+            replicaRequestReader,
+            executor,
+            syncGlobalCheckpointAfterOperation,
+            forceExecutionOnPrimary,
+            false
+        );
+    }
+
+    protected TransportReplicationAction(
```

Comment:
We've now got over 100 lines of overloaded constructors. Maybe it's time to replace this with the builder pattern? How feasible is that?

Reply:
Ack

Reply:
Certainly, we should move to Builder pattern here. However, i am bit skeptical on doing it in this CR due to indirect changes in multiple test cases that would get impacted. Created #5436. I will try to fix it in this PR, if possible. Otherwise, will do a follow up PR just after this one.

Reply:
Removed the constructor overloading with redesign in LLD. @andrross